core-application/src/main/java/gohigher/user/service/DesiredDesiredPositionCommandService.java

```diff
@@ -0,0 +1,60 @@
+package gohigher.user.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import gohigher.global.exception.GoHigherException;
+import gohigher.position.PositionErrorCode;
+import gohigher.position.port.out.PositionPersistenceQueryPort;
+import gohigher.user.port.in.DesiredPositionCommandPort;
+import gohigher.user.port.out.DesiredPositionPersistenceCommandPort;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class DesiredDesiredPositionCommandService implements DesiredPositionCommandPort {
+
+	private static final int MAIN_POSITION_IDX = 0;
+
+	private final PositionPersistenceQueryPort positionPersistenceQueryPort;
+	private final DesiredPositionPersistenceCommandPort desiredPositionPersistenceCommandPort;
+
+	@Override
+	public void saveDesiredPositions(Long userId, List<Long> positionIds) {
+		List<Long> distinctPositionIds = positionIds.stream()
+			.distinct()
+			.collect(Collectors.toList());
+		validateDuplicatedPositionIds(positionIds, distinctPositionIds);
+		validateExistedPositions(distinctPositionIds);
+
+		Long mainDesiredPositionId = extractMainPositionId(positionIds);
+		List<Long> subDesiredPositionIds = new ArrayList<>(distinctPositionIds);
+		desiredPositionPersistenceCommandPort.saveDesiredPositions(userId, mainDesiredPositionId,
+			subDesiredPositionIds);
+	}
+
+	private void validateDuplicatedPositionIds(List<Long> positionIds, List<Long> distinctPositionIds) {
+		if (distinctPositionIds.size() != positionIds.size()) {
+			throw new GoHigherException(PositionErrorCode.DUPLICATED_POSITION);
+		}
+	}
```

Comment:
1. 29~31번째 라인의 `distinctPositionIds`는 `saveDesiredPosition()`의 로직에서 필요한 데이터가 아니라 `validateDuplicatedPositionIds()`에 필요한 데이터가 아닌가?하는 생각이 듭니다. 
`validateDuplicatedPositionIds()`에서 중복된 id가 없다는 것이 검증이 됐으면 `validateExistedPositions()`에는 `positionIds`를 넘겨줘도 될 것 같아요!
2. 36번 라인의 `subDesiredPositionIds`는 List 객체를 새로 생성한 이유가 있을까요? 앞서 언급한 1번 내용과 같이 위에서 검증이 다 됐으면 `positionIds`와 같은 데이터가 만들어질 것 같은데 불필요한 선언인 것 같습니다.

```suggestion
	@Override
	public void saveDesiredPositions(Long userId, List<Long> positionIds) {
		Long mainDesiredPositionId = extractMainPositionId(positionIds);
		validateDuplicatedPositionIds(positionIds);
		validateExistedPositions(positionIds);
		desiredPositionPersistenceCommandPort.saveDesiredPositions(userId, mainDesiredPositionId, positionIds);
	}

	private void validateDuplicatedPositionIds(List<Long> positionIds) {
		List<Long> distinctPositionIds = positionIds.stream()
			.distinct()
			.collect(Collectors.toList());

		if (distinctPositionIds.size() != positionIds.size()) {
			throw new GoHigherException(PositionErrorCode.DUPLICATED_POSITION);
		}
	}
```

Reply:
1. 그러네요! 수정하겠습니다!!ㅎㅎ
2. 불필요한 선언인건 알지만 remove()로 인해 main이 빠졌다는 의미를 전달하고 싶었습니다!ㅜ 다른 방식으로 수정해보겠습니다!